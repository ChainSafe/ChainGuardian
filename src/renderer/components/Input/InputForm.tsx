import * as React from "react";

export interface IInputFormProps {
    label?: string;
    valid?: boolean;
    errorMessage?: string;
    inputValue?: any;
    placeholder?: string;
    onChange?: (e: React.FormEvent<HTMLInputElement>) => void;
    focused?: boolean;
    inputId?: string;
}

export const InputForm: React.FunctionComponent<IInputFormProps> = (props: IInputFormProps) => {
    
    const classNamesValid = (props: boolean | undefined): string => {
        switch(props) {
            case undefined : return("");
            case true : return("success");
            case false : return("error");
        }
    };

    return( 
        <form>
            <div className="label">{props.label}</div>
            <input 
                id={props.inputId}
                autoFocus={props.focused}
                placeholder={props.placeholder}
                value={props.inputValue}
                className={`inputform ${classNamesValid(props.valid)}`} 
                onChange={props.onChange} />
            <div 
                className={"error-message"}>
                {props.valid === false && props.errorMessage}</div>
        </form>
    );
};