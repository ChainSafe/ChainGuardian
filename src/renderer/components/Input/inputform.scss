.inputform::placeholder{
    color: $neutral-100;
}
.input-eye{
    margin-left: 12px;
    width: 50px;
    height: 50px;
    background: url(../assets/img/input/eye-regular.svg) no-repeat center center;
    &.none{
        display: none;
    }
    &:hover{
        background: url(../assets/img/input/eye-regular-hover.svg) no-repeat center center;
    }
}
.input-eye-slash{
    margin-left: 12px;
    width: 50px;
    height: 50px;
    background: url(../assets/img/input/eye-slash-regular.svg) no-repeat center center;
    &.none{
        display: none;
    }
    &:hover{
        background: url(../assets/img/input/eye-slash-regular-hover.svg) no-repeat center center;
    }
}
.inputform-container{
    width: 100%;
    margin-right: 24px;
    display: flex;
    align-items: center;

    .input-label {
        font-family: 'Palanquin', sans-serif;
        font-size: 18px;
        font-weight: normal;
        color: $neutral-300;
        z-index: 100;
    }
}
.inputform{
    width: 100%;
    background: $neutral-600;
    box-shadow: inset 0 2px 6px rgba(10, 39, 51, 0.2);
    border-radius: 4px;
    border: 2px solid $neutral-700;
    outline: transparent;
    font-family: 'Palanquin', sans-serif;
    font-size: 18px;
    color: $neutral-100;
    padding: 16px 56px 16px 16px;

    &:disabled {
        color: $neutral-400;
        padding-right: 16px;
    }
    &.success{
        background: $neutral-600 url(../assets/img/input/Success.svg) no-repeat;
        background-position: bottom 18px right 16px;
        border: 2px solid $system-success-500;
    }
    &.error{
        background: $neutral-600 url(../assets/img/input/Error.svg) no-repeat;
        background-position: bottom 18px right 16px;
        border: 2px solid $system-error-500;
    }
    &.centered {
        text-align: center;
    }
}

.label{
    font-family: 'Palanquin', sans-serif;
    font-size: 18px;
    color: $neutral-100;
    margin: 0px;
    padding: 0px;
}
.error-message{
    color: $system-error-300;
    font-size: 12px;
    height: 18px;
    margin: 0 18px;
    width: 442px;
    font-family: 'Lato', sans-serif;
    &.none{
        display: none;
    }
}