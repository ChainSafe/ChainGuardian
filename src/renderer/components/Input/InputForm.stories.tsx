import * as React from "react";
import {storiesOf} from "@storybook/react";
import {withKnobs, text, select} from "@storybook/addon-knobs";
import {InputForm} from "./InputForm";

storiesOf("Input Form", module)
    .add("Input Form", () => {
        const options = {
            gray: "",
            success: true,
            error: false,
        };
        const defaultValue: any = undefined;
        const label = text("label text", "Input");
        const validValue = select("valid", options, defaultValue);
        const eMessage = text("error message", "Error message");
        const idValue = text("id", "");
        return (
            <InputForm
                inputId={idValue}
                valid={validValue}
                label={label}
                errorMessage={eMessage}
                onSubmit={(): void => {}}
                eye
            />
        );
    })
    .addDecorator(withKnobs);
