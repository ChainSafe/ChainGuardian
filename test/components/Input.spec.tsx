import * as React from "react";
import * as renderer from "react-test-renderer";
import {InputForm} from "../../src/renderer/components/Input/InputForm";

describe("InputForm", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<InputForm label='Input' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders success", () => {
        const tree = renderer.create(<InputForm valid label='Input' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders error and message", () => {
        const tree = renderer.create(<InputForm valid={false} label='Input' errorMessage='Error message' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders value", () => {
        const tree = renderer.create(<InputForm inputValue='textfortest' label='Input' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused input", () => {
        const tree = renderer.create(<InputForm focused label='Input' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders input id", () => {
        const tree = renderer.create(<InputForm inputId='testid' label='Input' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
