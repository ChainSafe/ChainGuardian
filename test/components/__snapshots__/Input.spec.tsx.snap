// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders correctly 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    className="inputform "
  />
  <div
    className="error-message"
  />
</form>
`;

exports[`InputForm renders error and message 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    className="inputform error"
  />
  <div
    className="error-message"
  >
    Error message
  </div>
</form>
`;

exports[`InputForm renders focused input 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    autoFocus={true}
    className="inputform "
  />
  <div
    className="error-message"
  />
</form>
`;

exports[`InputForm renders input id 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    className="inputform "
    id="testid"
  />
  <div
    className="error-message"
  />
</form>
`;

exports[`InputForm renders success 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    className="inputform success"
  />
  <div
    className="error-message"
  />
</form>
`;

exports[`InputForm renders value 1`] = `
<form>
  <div
    className="label"
  >
    Input
  </div>
  <input
    className="inputform "
    value="textfortest"
  />
  <div
    className="error-message"
  />
</form>
`;
